src/screens/LoginScreen.tsx

```diff
@@ -48,20 +47,12 @@ const LoginScreen = ({
 
       setIsPinModalVisible(false)
 
-      dispatch(pinEntered(pin))
-      await dispatch(activeWalletChanged(wallet))
-
-      if (resetWalletOnLogin) {
-        dispatch(addressesFlushed())
-      }
-
-      if (resetWalletOnLogin || addressesStatus === 'uninitialized') {
-        dispatch(addressesFromStoredMetadataInitialized())
-      }
+      const requiresAddressInitialization = resetWalletOnLogin || addressesStatus === 'uninitialized'
+      await switchWallet(wallet, requiresAddressInitialization, pin)
 
       restoreNavigationState(resetWalletOnLogin)
     },
-    [addressesStatus, dispatch, resetWalletOnLogin, restoreNavigationState]
+    [addressesStatus, resetWalletOnLogin, restoreNavigationState, switchWallet]
```

Comment:
This goes in the direction of my previous argument about not getting `switchWallet` from a hook : IMO it makes no sense that `switchWallet` is part of the deps here, as the function won't change (it doesn't rely on state).

Reply:
Yes, it's there for the same reason `dispatch` was there.

Reply:
Probably fixed by https://github.com/alephium/mobile-wallet/commit/44ae680fe9934a1cb7474ff194e3aaf0b8051013